chia/full_node/mempool_manager.py

```diff
@@ -103,6 +116,81 @@ def compute_assert_height(
     return height
 
 
+def run_for_cost_and_additions(
+    coin_id: bytes32, puzzle_reveal: SerializedProgram, solution: SerializedProgram, maximum_cost: uint64
+) -> Tuple[uint64, List[Coin]]:
+    cost, conditions = puzzle_reveal.run_with_cost(maximum_cost, solution)
+    created_coins = []
+    for condition in conditions.as_python():
```

Comment:
it's not safe to use `as_python()`. It will (recursively) convert the *entire* lisp tree into python objects. If the puzzle is malicious, that's more than you have RAM.

Instead, you need to iterate the list of `SExp` objects, pairs and atoms. You could try to call `as_iter()` on the `SExp` object. It will give you an iterator over `SExp` items in the list.